### DISCUSSION TITLE
Verständnissfragen zum Code

### DISCUSSION THREAD

**Discussion by DOHRMANN-SOFTWARE** [scheb/open-apexx#63] [General]
Da ich immer mehr auf Probleme stoße, kommen hier ein paar Fragen.

Kann es sein, das du nach und nach Funktionen eingebaut hast, die im Grunde auf der selben Art das Endergebniss ausgeben, aber anders zu handhaben sind.

Man kann ja Produkte mit einem anderen Produkt verknüpfen.

Das machst du bei dem bearbeiten eines Produkts, einmal mit `$apx->tmpl->assign('PRODID',intval($_POST['prodid']));` und im Template mit `{PRODUCTS(PRODID,ID)}`, damit man es in der Auswahllist auswälen kann.

Aber bei einem Hersteller machst du das mit `$apx->tmpl->assign('MANUFACTURER',$this->get_units($_POST['manufacturer'], iif($set['products']['filtermanu'], 'person'))); und im` Template mit `{MANUFACTURER}`.

Bei beiden wird aus einer Liste eine wahl getroffen.
Das die Daten zum Teil unterschiedlich aus der Datenbank gelesern werden muss, ist mir schon klar.
Aber könnte man die Ausgabe der Daten für die Auswahlliste, nicht auf die slebe Art ausgeben?
Oder versteh ich wieder was falsch?

Kann du mir sagen, wo die Funktion zu `{PRODUCTS(PRODID,ID)}` zu finden ist?

Wieso existiert die Datei `admin_system.php`, hatte mann den Inhalt nicht in die Datei `admin_extend.php` oder in die Datei `functions.php` packen können?

**Comment by scheb** [comment]
> Kann es sein, das du nach und nach Funktionen eingebaut hast, die im Grunde auf der selben Art das Endergebniss ausgeben, aber anders zu handhaben sind.
> [...]

Kann sein, aber den Hintergrund warum das so ist, kann ich dir nicht mehr sagen. Das ist zu lange her. Evtl. haben sich da im Laufe der Entwicklung neue Implementierungs-Möglichkeiten ergeben, aber das ist nur eine Vermutung.

> Kann du mir sagen, wo die Funktion zu {PRODUCTS(PRODID,ID)} zu finden ist?

Also laut dieser Zeile hier https://github.com/scheb/open-apexx/blob/master/src/modules/products/init.php#L71 ist das die Funktion `products_list`. Musst du mal schauen wo im Quelltext die sich versteckt.

> Wieso existiert die Datei admin_system.php, hatte mann den Inhalt nicht in die Datei admin_extend.php oder in die Datei functions.php packen können?

Welchen Zweck die einzelnen Dateien haben kann ich nicht mehr sagen. Soweit ich mich erinnern kann werden unterschiedliche Dateien zu unterschiedlichen Zeitpunkten geladen, bzw. gewisse Dateien (bei der tfunctions.php bin ich mir ziemlich sicher) werden nur geladen wenn daraus etwas benötigt wird.
